### ISSUE TITLE
[ERROR] [develop] Builder is already started on remote ssh linux

### ISSUE THREAD

**Issue by ccecvb** [vscode-abl/vscode-abl#167]
@gquerret  I created separate issue to a answer your question in #165,
I'm not sure I can reliably reproduce the issue. I got the error 2 times out of 3 in the sequence below.

* vscode on windows
* connected over ssh to linux
* OpenEdge abl extension v1.9.16 (pre-release) on linux
* Linux Red Hat Enterprise Linux release 9.4 (Plow)

The problem occurs after 

vscode  -> file -> close workspace 

followed by 

File -> Open recent -> select same workspace

* pid in builder.lock not found

```
# cat .builder/builder.lock 
391702
# ps -fp 3917027
UID          PID    PPID  C STIME TTY          TIME CMD

```

If I repeat the close+open recent sequence I notice that the pid in .builder/builder.lock is updated but that pid does not exist either.
But this time the language server does NOT show any errors

```
# cat .builder/builder.lock 
4082307
# ps -fp 4082307
UID          PID    PPID  C STIME TTY          TIME CMD

```
3rd attempt fails again, pid is still the same as above.

Languahe server output of first open recent

```
[2024-07-26T07:52:18.965+0200] [INFO] Command line arguments: 
[2024-07-26T07:52:19.040+0200] [INFO] vscode.rootDir: /home/cvb@CCE.local/.vscode-server/extensions/riversidesoftware.openedge-abl-lsp-1.9.16-linux-x64/resources
[2024-07-26T07:52:19.272+0200] [INFO] LanguageServerInfo: 1.15.0-SNAPSHOT-68539a1 -- ClientInfo: Visual Studio Code 1.91.1 -- JVM: Eclipse Adoptium 17.0.11+9
[2024-07-26T07:52:19.272+0200] [INFO] Client Capabilities: WS Config true -- WS Folder true -- Diagnotics true -- Symbol Hierarchy true -- Document Highlight true
[2024-07-26T07:52:19.272+0200] [INFO] Semantic tokens support: true - Multiline tokens: false - Overlapping tokens: false
[2024-07-26T07:52:19.312+0200] [INFO] Register CABL License for 'Claerhout Computer Engineering nv' - Valid until 2024-10-01T00:00:00
[2024-07-26T07:52:19.316+0200] [INFO] Looking for project JSON in workspace folder file:///home/cvb%40CCE.local/workspaces/develop
[2024-07-26T07:52:19.339+0200] [INFO] [develop] Current profile: default
[2024-07-26T07:52:19.339+0200] [INFO] [develop] Progress version : 12.8 -> /usr/oe128
[2024-07-26T07:52:19.362+0200] [INFO] [develop] Source directory /home/cvb@CCE.local/workspaces/develop/git/ccesmarttools/CceFramework/src
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> Build into /home/cvb@CCE.local/workspaces/develop/tty
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> PCT dir    /home/cvb@CCE.local/workspaces/develop/.builder/.pct0
[2024-07-26T07:52:19.363+0200] [INFO] [develop] Source directory /home/cvb@CCE.local/workspaces/develop/git/smartcomponentlibrary
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> Build into /home/cvb@CCE.local/workspaces/develop/tty
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> PCT dir    /home/cvb@CCE.local/workspaces/develop/.builder/.pct1
[2024-07-26T07:52:19.363+0200] [INFO] [develop] Source directory /home/cvb@CCE.local/workspaces/develop/git/smartcomponentlibrary/src
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> Build into /home/cvb@CCE.local/workspaces/develop/tty
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> PCT dir    /home/cvb@CCE.local/workspaces/develop/.builder/.pct2
[2024-07-26T07:52:19.363+0200] [INFO] [develop] Source directory /home/cvb@CCE.local/workspaces/develop/git/lisaerp/appl/src
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> Build into /home/cvb@CCE.local/workspaces/develop/tty
[2024-07-26T07:52:19.363+0200] [INFO] [develop]   --> PCT dir    /home/cvb@CCE.local/workspaces/develop/.builder/.pct3
[2024-07-26T07:52:19.363+0200] [INFO] [develop] Source directory /home/cvb@CCE.local/workspaces/develop/git/lisafin/applfin/src
[2024-07-26T07:52:19.364+0200] [INFO] [develop]   --> Build into /home/cvb@CCE.local/workspaces/develop/tty
[2024-07-26T07:52:19.364+0200] [INFO] [develop]   --> PCT dir    /home/cvb@CCE.local/workspaces/develop/.builder/.pct4
[2024-07-26T07:52:19.364+0200] [INFO] [develop] PROPATH recap:
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   git/ccesmarttools/CceFramework -> /home/cvb@CCE.local/workspaces/develop/git/ccesmarttools/CceFramework
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   git/ccesmarttools/CceFramework/src -> /home/cvb@CCE.local/workspaces/develop/git/ccesmarttools/CceFramework/src
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   git/smartcomponentlibrary -> /home/cvb@CCE.local/workspaces/develop/git/smartcomponentlibrary
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   git/smartcomponentlibrary/src -> /home/cvb@CCE.local/workspaces/develop/git/smartcomponentlibrary/src
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   git/lisaerp/appl/src -> /home/cvb@CCE.local/workspaces/develop/git/lisaerp/appl/src
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   git/lisafin/applfin/src -> /home/cvb@CCE.local/workspaces/develop/git/lisafin/applfin/src
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   oe128/oecommon/ade/src -> /home/cvb@CCE.local/workspaces/develop/oe128/oecommon/ade/src
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   /usr/oe128 -> /usr/oe128
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   /usr/oe128/bin -> /usr/oe128/bin
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   /usr/oe128/tty -> /usr/oe128/tty
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   /usr/oe128/tty/OpenEdge.Core.pl -> /usr/oe128/tty/OpenEdge.Core.pl
[2024-07-26T07:52:19.365+0200] [INFO] [develop]   /usr/oe128/tty/netlib/OpenEdge.Net.pl -> /usr/oe128/tty/netlib/OpenEdge.Net.pl
[2024-07-26T07:52:20.307+0200] [INFO] Project found
[2024-07-26T07:52:20.311+0200] [ERROR] [develop] Builder is already started
[2024-07-26T07:52:20.331+0200] [INFO] Looking for ABL documentation cache in /home/cvb@CCE.local/.rssw/oedoc.bin

```

**Comment by ccecvb** [comment]
A colleague had a variation of this issue
The same` builder is already started` but in his case there was a process with the pid in the .builder.lock file.

killing that process and restarting the language server fixed the issue.

Config similar as described above but running from a Macbook instead of a Windows laptop and not using the preview version of the plugin

**Comment by gquerret** [comment]
Fixed in 1.9.18 (will be in the marketplace soon). There was a thread synchronization issue, so you didn't see the problem every time.
